tests/behaviour/typeql/steps.rs

```diff
@@ -20,49 +20,72 @@
  */
 
 use cucumber::{gherkin::Step, given, then, when};
-use futures::TryStreamExt;
+use futures::{TryFutureExt, TryStreamExt};
+use typeql_lang::parse_query;
 
 use crate::{behaviour::Context, generic_step_impl};
 
 generic_step_impl! {
     #[step(expr = "typeql define")]
     async fn typeql_define(context: &mut Context, step: &Step) {
-        // FIXME parse
-        context.transaction().query().define(step.docstring().unwrap()).await.unwrap();
+        let parsed = parse_query(step.docstring().unwrap());
+        assert!(parsed.is_ok());
+        let res = context.transaction().query().define(&parsed.unwrap().to_string()).await;
+        assert!(res.is_ok());
     }
 
     #[step(expr = "typeql define; throws exception")]
     async fn typeql_define_throws(context: &mut Context, step: &Step) {
-        // FIXME parse
-        assert!(context.transaction().query().define(step.docstring().unwrap()).await.is_err());
+        let parsed = parse_query(step.docstring().unwrap());
+        if parsed.is_ok() {
+            let res = context.transaction().query().define(&parsed.unwrap().to_string()).await;
+            assert!(res.is_err());
+        }
     }
 
     #[step(expr = "typeql define; throws exception containing {string}")]
     async fn typeql_define_throws_exception(context: &mut Context, step: &Step, exception: String) {
-        // FIXME parse
-        let res = context.transaction().query().define(step.docstring().unwrap()).await;
-        assert!(res.is_err());
-        assert!(res.unwrap_err().to_string().contains(&exception));
+        let result = async { parse_query(step.docstring().unwrap()).map_err(|error| error.to_string()) }
+            .and_then(|parsed| async move {
+                context.transaction().query().define(&parsed.to_string()).await.map_err(|error| error.to_string())
+            })
+            .await;
```

Comment:
This async / `.map_err` / `.and_then` chaining warrants a comment for future reference, even though we've talked through it on a call.

Reply:
We replaced `match`es with the combinators chaining to avoid repetitive exception checks.
We convert all errors (of incompatible types) to `String`s because they must have the same type.
The result of query parsing must be a `Future` because we apply the `async` function (using `and_then`) that returns a `Future` to it.